enrichments/trace/internal/elastic/span.go

```diff
@@ -237,6 +238,39 @@ func (s *spanEnrichmentContext) enrichTransaction(
 	}
 }
 
+// In OTel a root span can represent an outgoing call or a producer span
+// In such cases, the span is still mapped into a transaction, but such spans are enriched
+// with additional attributes that are specific to the outgoing call or producer span.
+func (s *spanEnrichmentContext) enrichExitSpanTransaction(
+	span ptrace.Span,
+	cfg config.Config,
+) {
+	if span.Kind() == ptrace.SpanKindClient || span.Kind() == ptrace.SpanKindProducer {
+		if cfg.Span.TypeSubtype.Enabled {
+			s.setSpanTypeSubtype(span)
+		}
+		if cfg.Span.ServiceTarget.Enabled {
+			s.setServiceTarget(span)
+		}
+		if cfg.Span.DestinationService.Enabled {
+			s.setDestinationService(span)
+		}
+		if cfg.Span.Name.Enabled {
+			span.Attributes().PutStr(AttributeSpanName, span.Name())
+		}
+		if cfg.Transaction.Type.Enabled {
+			spanTypeAttr, hasType := span.Attributes().Get(AttributeSpanType)
+			if hasType {
+				transactionType := spanTypeAttr.Str()
+				if spanSubtypeAttr, hasSubType := span.Attributes().Get(AttributeSpanSubtype); hasSubType {
+					transactionType += "." + spanSubtypeAttr.Str()
```

Comment:
I'd be happy to hear opinion on this one.

The APM UI groups transactions based on transaction type. The idea here is that for such exit&root spans we'd want to use a different transaction type so they don't end up in the same group with incoming transactions.

<img width="458" alt="Screenshot 2024-11-28 at 16 55 18" src="https://github.com/user-attachments/assets/de8679e9-8335-4721-9c7a-cb079d856dcc">

E.g. the default transaction type for incoming HTTP is `request` - with this, for outgoing we'd use `external.http`. It can be that a service sends `GET` both as outgoing root span and as an incoming transaction. This way those 2 transactions will have different types.

Alternatively to `.` we could also have `/` - e.g. `external/http`.

Reply:
will that impact other things like service map or so?

Reply:
So far, I haven't seen any impact. Service map is mostly driven by the `target.*` and `destination.*` fields from what I know.

The dependency list has icons - that is based on this span, but that is not driven by `transaction.type` either, so that's not impacted either.